design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,271 @@
+     1. *Error code 1*: The requesting user is forbidden from accessing the configurations or functionality (like generating credentials).
+  2. 400- Bad request: 
+     1. *Error code 1*: The request is missing information necessary for the template’s expansion.
+  3. 500- Internal server error: 
+     1. *Error code 1*: The lakeFS server cannot access some of the provided template locations.
+
+---
+
+## Possible Flows
+
+### Example template
+
+**Databricks Spark configurations**  
+Location: s3://org-super-bucket/spark/databricksConfig.props
+```properties
+spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
+{{with $creds := new_credentials}}
+spark.hadoop.fs.s3a.access_key={{$creds.Key}}
+spark.hadoop.fs.s3a.secret_key={{$creds.Secret}}
+{{end}}
+spark.hadoop.fs.s3a.endpoint={{ .querystring.lakefs_url }}
+spark.hadoop.fs.s3a.path.style.access=true
+```
+
+### Happy flow - All steps
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to type the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user specifies the location of the bucket (after they validated that the lakeFS role has the right permissions and that the bucket has the correct policy), and clicks ‘OK’.
+   1. A progress bar or counter will show the progress of the import process and will signal once it’s over.
+5. The wizard will ask the user: “How are you using Spark”, and will offer three alternatives: Databricks, AWS EMR and spark-submit. The user chooses ‘Databricks’. It also asks the user for their lakeFS endpoint (and will show a default placeholder pointing to the current URL). The user types the endpoint.
+6. The wizard will send a GET request to the templating service with a query string of the format:  
+    ```
+    ?lakefs_url=<url>&template_location=s3://org-super-bucket/spark/databricksConfig.props
+    ```
+7. The templating service will fetch the templates from the provided locations, expand the `.querystring.lakefs_url` parameter for both templates and return a json response of the format:
+    ```json
+    {
+        "s3://org-super-bucket/spark/databricksConfig.props": {
+            "expended": [the expanded Spark template as a string],
+            "secrets": [“<secret1>”, “<secret2>”...]
+        }
+    }
+    ```
+8. The wizard will present each configuration in a different snippet view for the users to copy and paste into their configuration files.
+9. The wizard will generate a README file as follows:
+    ```markdown
+    1. Created a repository "spark-repo" and branch "main".
+    2. Imported data from <S3 location>.
+    3. Generated the following configurations:
+       <Spark configurations with [“<secret1>”, “<secret2>”...] hidden>
+    4. Instructions to configure Hive Metastore with lakeFS.
```

Comment:
And, which `lakectl metastore` commands to run when creating/deleting/merging a branch/tag.

Reply:
Point one will generate configurations that are exactly the same as the Spark configurations generated...